src/WWT.Providers/Providers/DSSProvider.cs

```diff
@@ -1,70 +1,65 @@
 using System;
 using System.Configuration;
-using System.IO;
 using WWTWebservices;
 
 namespace WWT.Providers
 {
     public class DSSProvider : RequestProvider
     {
-        public override void Run(WwtContext context)
+        private readonly IPlateTilePyramid _plateTile;
+
+        public DSSProvider(IPlateTilePyramid plateTile)
         {
-            string wwtTilesDir = ConfigurationManager.AppSettings["WWTTilesDir"];
-            string dssTerapixelDir = ConfigurationManager.AppSettings["DssTerapixelDir"];
+            _plateTile = plateTile;
+        }
 
+        public override void Run(WwtContext context)
+        {
             string query = context.Request.Params["Q"];
             string[] values = query.Split(',');
             int level = Convert.ToInt32(values[0]);
             int tileX = Convert.ToInt32(values[1]);
             int tileY = Convert.ToInt32(values[2]);
 
-            int octsetlevel = level;
-            string filename;
-
-
             if (level > 12)
             {
                 context.Response.Write("No image");
                 context.Response.Close();
-                return;
             }
-
-            if (level < 8)
+            else if (level < 8)
             {
+                string wwtTilesDir = ConfigurationManager.AppSettings["WWTTilesDir"];
+
                 context.Response.ContentType = "image/png";
-                Stream s = PlateTilePyramid.GetFileStream(wwtTilesDir + "\\dssterrapixel.plate", level, tileX, tileY);
-                int length = (int)s.Length;
-                byte[] data = new byte[length];
-                s.Read(data, 0, length);
-                context.Response.OutputStream.Write(data, 0, length);
-                context.Response.Flush();
-                context.Response.End();
-                return;
+
+                using (var s = _plateTile.GetStream(wwtTilesDir, "dssterrapixel.plate", level, tileX, tileY))
+                {
+                    s.CopyTo(context.Response.OutputStream);
+                    context.Response.Flush();
```

Comment:
Previously, the stream was copied into memory and then written out. There is no need to do this as it can be streamed with the `.CopyTo(Stream)` method (which was probably not around when this was first written).